profiles/views.py

```diff
@@ -194,8 +197,14 @@ def get_serializer_class(self):
     def perform_destroy(self, instance):
         serializer = self.get_serializer(data=self.request.data)
         serializer.is_valid(raise_exception=True)
-        instance.is_deleted = True
-        instance.save()
+        user = self.request.user
+        with transaction.atomic():
+            instance.is_deleted = True
+            instance.save()
+            user.is_active = False
+            user.email = f"is_deleted_{now()}_{user.email}"
```

Comment:
I recommend to specify datetime format here. To avoid unnecessary unpredicted issues.

With current implementation, you'll have spaces, `:`, `.`, `+` and God knows what else, + even the length of this can be different for different cases (for example, normally you'll have 6-digits representation of microseconds, but if microseconds were equal to `000000` — they will be omitted in string representation of datetime.

So, `.strftime()` returning timestamp in format `YYYYMMDDhhmmss` would be a cleaner/safer solution.

Also, I'd strongly recommend to have a unittest for this case (checking specific values, by patching `now()` function).

Reply:
Done